api/proto/auth/client.go

```diff
@@ -0,0 +1,889 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package auth implements teleport's grpc auth client
+package auth
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"io"
+	"net"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/gravitational/roundtrip"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/jwt"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/session"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/trace/trail"
+	"golang.org/x/net/http2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	ggzip "google.golang.org/grpc/encoding/gzip"
+	"google.golang.org/grpc/keepalive"
+)
+
+func init() {
+	if err := ggzip.SetLevel(gzip.BestSpeed); err != nil {
+		panic(err)
+	}
+}
+
+// Client is a gRPC Client that connects to a teleport auth server through TLS.
+type Client struct {
+	c    Config
+	grpc AuthServiceClient
+	conn *grpc.ClientConn
+	// closedFlag is set to indicate that the services are closed
+	closedFlag int32
+}
+
+// TLSConfig returns TLS config used by the client, could return nil
+// if the client is not using TLS
+func (c *Client) TLSConfig() *tls.Config {
+	return c.c.TLS
+}
+
+// NewTLSClient returns a new auth client that uses mutual TLS authentication and
+// connects to the remote server using the Dialer or Addrs in Config.
+func NewTLSClient(cfg Config, params ...roundtrip.ClientParam) (*Client, error) {
```

Comment:
Does this code use `roundrip.ClientParam` here?
We can name it `NewClient`, `TLS` is the only option in this package.